src/cuda/tests.rs

```diff
@@ -0,0 +1,290 @@
+use crate::cuda::*;
+
+static CORRECT_ENCODING: [u8; 4096] = [
```

Comment:
This is identical to CPU, would be nice to put it into shared private module instead of copy-pasting.